<a href="https://colab.research.google.com/github/JITHINKVT/AI_ML_COURSE_ICTAK_JITHIN/blob/main/Manappuram_Python_Assgnt1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

1. A financial analyst stores monthly revenues in a list: [120000, 135000, 128000,
142000, 150000]. Find the difference between the highest and lowest revenue without
using loops

In [ ]:
rev = [120000, 135000, 128000, 142000, 150000]
diff = max(rev) - min(rev)
print(diff)

30000


Ans.)30000

2. Given the tuple account = ('Asha', 'Savings', 56000, 'KCH'), unpack the values into
separate variables and print a formatted string: 'Asha has ₹56000 in her Savings
account (Branch: KCH)'.

In [ ]:
acc = ('Asha', 'Savings', 56000, 'KCH')
Name = acc[0]
Acc_type = acc[1]
Bal_amount = acc[2]
Br_name = acc[3]
print(f'{Name} has ₹{Bal_amount} in her {Acc_type} account (Branch: {Br_name})')

Asha has ₹56000 in her Savings account (Branch: KCH)


Ans.)Asha has ₹56000 in her Savings account (Branch: KCH)

3. Convert the list of string numbers ['45000', '56000', '38000'] to integers and find
their average balance.

In [ ]:
str_num = ['45000', '56000', '38000']
int_num = [int(num) for num in str_num]
avg_bal = sum(int_num) / len(int_num)
print(f"The average balance is: {avg_bal}")

The average balance is: 46333.333333333336


Ans.)The average balance is: 46333.333333333336


4. Create a list with elements: ['Kochi', 'Chennai', 'Mumbai', 'Delhi', 'Pune']. Replace
'Delhi' with 'Hyderabad' and print the new list in reverse order using slicing.

In [ ]:
city=['Kochi','Chennai','Mumbai','Delhi','Pune']
city[-2]='Hyderabad'
print(city[::-1])

['Pune', 'Hyderabad', 'Mumbai', 'Chennai', 'Kochi']


Ans.)['Pune', 'Hyderabad', 'Mumbai', 'Chennai', 'Kochi']

5. You have two tuples: interest_2024 = (7.1, 7.3, 7.2) and interest_2025 = (7.4, 7.5).
Combine them and print the average of all interest rates.

In [ ]:
int_2024 = (7.1, 7.3, 7.2)
int_2025 = (7.4, 7.5)
all_interest = int_2024 + int_2025
avg_interest = sum(all_interest) / len(all_interest)
print(f"The average interest rate is: {avg_interest}")

The average interest rate is: 7.3


Ans.)The average interest rate is: 7.3

6. A string variable branch_id = 'TVM-FIN-2025'. Extract the branch code (first three
letters) and the year (last four digits)

In [ ]:
br_id = 'TVM-FIN-2025'
br_code = br_id[:3]
year = br_id[-4:]
print(f"Branch Code: {br_code}")
print(f"Year: {year}")

Branch Code: TVM
Year: 2025


Ans.)Branch Code: TVM
Year: 2025

7. Create a list transactions = [10000, -2000, 5000, -1500, 3000]. Without using loops,
find total deposits and withdrawals using list slicing or indexing only

In [ ]:
t1 = [10000, -2000, 5000, -1500, 3000]
total_depo = sum(t1[::2])
total_withdraw = sum(t1[1::2])
print(f"Total Deposits: {total_depo}")
print(f"Total Withdrawals: {total_withdraw}")


Total Deposits: 18000
Total Withdrawals: -3500


Ans.)Total Deposits: 18000
Total Withdrawals: -3500

8. Given name = 'John Doe' and balance = 72500, print a message in title case stating
'John Doe has a balance of ₹72500'. Ensure no type errors occur.

In [ ]:
name = 'John Doe'
balance = 72500
print(f"{name.title()} has a balance of ₹{balance}")


John Doe has a balance of ₹72500


Ans.)John Doe has a balance of ₹72500

9. A tuple customer = ('Kiran', 'Loan', 80000, 7.5). Calculate and print the total amount
payable after 1 year if interest is simple interest based on rate 7.5%

In [ ]:
cust = ('Kiran', 'Loan', 80000, 7.5)
calc = cust[2] * (1 + (cust[3] / 100))
print(f"The total amount payable after 1 year is: {calc}")


The total amount payable after 1 year is: 86000.0


Ans.)The total amount payable after 1 year is: 86000.0

10. Create two lists – branches = ['Kochi', 'Chennai', 'Mumbai'] and profits = [120000,
95000, 134000]. Form a new list of tuples pairing each branch with its profit

In [ ]:
Branch = ['Kochi', 'Chennai', 'Mumbai']
Profit = [120000, 95000, 134000]
New_list = [(Branch[0], Profit[0]),(Branch[1], Profit[1]),(Branch[2], Profit[2])]
print(New_list)

[('Kochi', 120000), ('Chennai', 95000), ('Mumbai', 134000)]


Ans.)[('Kochi', 120000), ('Chennai', 95000), ('Mumbai', 134000)]

11. From a list salaries = [45000, 56000, 38000, 62000, 54000], extract the top three
salaries using slicing only.

In [ ]:
sal = [45000, 56000, 38000, 62000, 54000]
sal.sort()
top_three = sal[-3:]
print(top_three)

[54000, 56000, 62000]


Ans.)[54000, 56000, 62000]

12. Given the tuple exchange_rates = (83.2, 83.1, 83.5, 83.0, 82.9), convert it into a list
and change the third value to 84.0.

In [ ]:
ex_rate = (83.2, 83.1, 83.5, 83.0, 82.9)
l1 = list(ex_rate)
l1[2] = 84.0
print(l1)


[83.2, 83.1, 84.0, 83.0, 82.9]


Ans.)[83.2, 83.1, 84.0, 83.0, 82.9]

13. Combine two lists customer_names = ['Asha', 'John', 'Meera'] and balances = [56000,
72000, 65000] into a formatted string such as 'Asha:56000, John:72000, Meera:65000'
without loops

In [ ]:
cust_name = ['Asha', 'John', 'Meera']
bal = [56000,72000, 65000]
print(cust_name[0],":",bal[0]," , ",cust_name[1],":",bal[1]," , ",cust_name[2],":",bal[2])

Asha : 56000  ,  John : 72000  ,  Meera : 65000


Ans.)Asha : 56000  ,  John : 72000  ,  Meera : 65000

14. Given account_info = (Deepa, 'Savings', 98000), use string formatting to print a
sentence 'Account Holder Deepz has ₹98000 in Savings account

In [ ]:
acc_holder = ('Deepa', 'Savings', 98000)
print(f"Account Holder {acc_holder[0]} has ₹{acc_holder[2]} in {acc_holder[1]} account")


Account Holder Deepa has ₹98000 in Savings account


Ans.)Account Holder Deepa has ₹98000 in Savings account

15. Create a nested list storing branch names and their monthly collections: [['Kochi',
[120000, 135000]], ['Chennai', [95000, 110000]]]. Print only the collection of Chennai
branch without using loops.

In [ ]:
Branch=['Kochi',[12000,135000]],['Chennai',[95000,110000]]
C_branch=Branch[1][1]
print(f'The collection of chennai branch is {C_branch}')

The collection of chennai branch is [95000, 110000]


Ans.)The collection of chennai branch is [95000, 110000]